python_modules/libraries/dagster-teradata/dagster_teradata/resources.py

```diff
@@ -0,0 +1,464 @@
+            )
+            if self._resolved_config_dict.get(k) is not None
+        }
+        return conn_args
+
+    @classmethod
+    def _is_dagster_maintained(cls) -> bool:
+        return True
+
+    @public
+    @contextmanager
+    def get_connection(self) -> Iterator[Union[SqlDbConnection, teradatasql.TeradataConnection]]:
+        teradata_conn = teradatasql.connect(
+            host=self.host,
+            user=self.user,
+            password=self.password,
+            database=self.database,
+        )
+        yield teradata_conn
+
+    def get_object_to_set_on_execution_context(self) -> Any:
+        # Directly create a TeradataConnection here for backcompat since the TeradataConnection
+        # has methods this resource does not have
+        return TeradataConnection(
+            config=self._resolved_config_dict,
+            log=get_dagster_logger(),
+            teradata_connection_resource=self,
+        )
+
+
+class TeradataConnection:
+    """A connection to Teradata that can execute queries. In general this class should not be
+    directly instantiated, but rather used as a resource in an op or asset via the
+    :py:func:`teradata_resource`.
+
+    Note that the TeradataConnection is only used by the teradata_resource. The Pythonic TeradataResource does
+    not use this TeradataConnection class.
+    """
+
+    def __init__(
+        self, config: Mapping[str, str], log, teradata_connection_resource: TeradataResource
+    ):
+        self.teradata_connection_resource = teradata_connection_resource
+        self.compute_cluster_manager = TeradataComputeClusterManager(self, log)
+        self.log = log
+
+    @public
+    @contextmanager
+    def get_connection(self) -> Iterator[Union[SqlDbConnection, teradatasql.TeradataConnection]]:
+        """Gets a connection to Teradata as a context manager.
+
+        If using the execute_query, execute_queries, or load_table_from_local_parquet methods,
+        you do not need to create a connection using this context manager.
+
+        Args:
+            raw_conn (bool): If using the sqlalchemy connector, you can set raw_conn to True to create a raw
+                connection. Defaults to True.
+
+        Examples:
+            .. code-block:: python
+
+                @op(
+                    required_resource_keys={"teradata"}
+                )
+                def get_query_status(query_id):
+                    with context.resources.teradata.get_connection() as conn:
+                        # conn is a Teradata Connection object or a SQLAlchemy Connection if
+                        # sqlalchemy is specified as the connector in the Teradata Resource config
+
+                        return conn.get_query_status(query_id)
+
+        """
+        with self.teradata_connection_resource.get_connection() as conn:
+            yield conn
+
+    @public
+    def execute_query(
+        self,
+        sql: str,
+        fetch_results: bool = False,
```

Comment:
If fetch_results false, execute_query doesn't return anything? So, one of the use case is, this method will be used queries that doesn't return any result like DROP table. Is this correct?